Chapter4 의존성자동주입

Min-Su opened the pull request:
#3

Min-Su committed:
-- coding convention 정리

Min-Su committed:
-- Chapter 4 의존자동주입
멤버, 메소드에 @Autowired 어노테이션 부여로 의존 자동주입

Min-Su committed:
-- Chapter 4 의존성자동주입
빈 생성 클래스에 @Autowired 적용
Config 클래스인 AppCtx 클래스 의존 주입 코드 수정

Min-Su committed:
-- Chapter 4 의존자동주입
같은 타입의 빈 객체가 여러개 있다면 스프링 컨테이너가 빈 생성시 에러 발생으로
예외처리된다.
이 때, 빈 객체와 주입 대상 객체에 @Qualifier 어노테이션 설정으로
주입될 의존 객체를 선택한다.

Min-Su committed:
-- Chapter4 의존성 자동주입
자동 주입과 명시적 의존 주입 간의 관계
수동으로 의존성을 주입해도 @Autowired 어노테이션이 우선됨

monorisk reviewed (commented):
금번에 개발한 코드들을 생성자 주입 방식으로 모두 변경해봅시다.

Min-Su committed:
-- Chapter4 의존자동주입
Chapter3, Chapter4 리뷰사항 수정

Min-Su reviewed (commented):
챕터3 이셉션 TC관련 없는 email로 호출하여 이셉션으로 던졌으나 테스트코드 진행시 이셉션으로 던져저 테스트코드가 의도는 알맞으나 테스트 코드 자체는 실패로 나오는 경우 알맞는 테스트의 방법인지 아니면 테스트 코드는 무조건 정상실행으로 끝나야 하는지 궁금합니다.

Min-Su committed:
-- Chapter5 컴포넌트 스캔
빈으로 등록될 클래스를 지정하는 @Component, @Component로 지정된 빈 클래스를 스캔하는 @ComponentScan
@ComponentScan 속성 값으로 basePackage = {"패키지경로"} 를 부여 할 수 있다.

Min-Su committed:
-- Chapter5 컴포넌트 스캔
@ComponentScan 어노테이션은 excludeFilters 속성을 주어 컴포넌트 스캔 대상을 제외하거나 포함할 수 있다
1) FilterType.REGEX - 정규표현식을 사용하여 제외 대상 지정
2) FilterType.ASPECTJ - AspectJ 패턴을 사용하여 제외 대상 지정
3) FilterType.ANNOTATION - 특정 어노테이션을 붙인 타입을 제외 대상 지정
4) FilterType.ASSIGNABLE_TYPE - 특정 타입이나 그 하위 타입을 제외대상 지정
*설정할 필터가 두 개 이상이면 속성에 배열 사용가능

Min-Su reviewed (commented):
#5
